src/electionguard_gui/services/key_ceremony_stages/key_ceremony_s1_join_service.py

```diff
@@ -0,0 +1,45 @@
+from os import getcwd, path
+from pymongo.database import Database
+
+from electionguard import to_file
+from electionguard.guardian import Guardian
+from electionguard_gui.models.key_ceremony_dto import KeyCeremonyDto
+from electionguard_gui.services.key_ceremony_stages.key_ceremony_stage_base import (
+    KeyCeremonyStageBase,
+)
+from electionguard_gui.services.guardian_service import make_guardian
+from electionguard_tools import GUARDIAN_PREFIX
+
+
+class KeyCeremonyS1JoinService(KeyCeremonyStageBase):
+    """Responsible for stage 1 of the key ceremony where guardians join"""
+
+    def run(self, db: Database, key_ceremony: KeyCeremonyDto) -> None:
+        key_ceremony_id = key_ceremony.id
+        user_id = self._auth_service.get_user_id()
+        self._key_ceremony_service.append_guardian_joined(db, key_ceremony_id, user_id)
+        # refresh key ceremony to get the list of guardians with the authoritative order they joined in
+        key_ceremony = self._key_ceremony_service.get(db, key_ceremony_id)
+        guardian_number = self._key_ceremony_service.get_guardian_number(
```

Comment:
the get_guardian_number function could raise an exception.  Do we want to add try/except handling at this level so that the error can be shown nicely on the GUI?